src/main/java/com/squareup/kotlinpoet/TypeSpec.kt

```diff
@@ -111,20 +111,20 @@ class TypeSpec private constructor(builder: TypeSpec.Builder) {
           codeWriter.emit(")")
         }
 
-        val extendsTypes: List<TypeName>
-        val implementsTypes: List<TypeName>
+        val superTypes: List<Pair<TypeName, String>>
```

Comment:
Maybe consider `List<CodeBlock>` and do `CodeBlock.of("%T", type)` or `CodeBlock.of("%T()", type)`.